src/components/navbar/navbar.js

```diff
@@ -2,29 +2,28 @@ import React, { useState, useEffect } from 'react';
 import './navbar.css';
 import { Link, useNavigate } from 'react-router-dom';
 import { useAuth } from '../../context/authContext';
+import axios from 'axios';
 
 export default function Navbar() {
     const [loginBtn, setLoginBtn] = useState();
-    const { user, setUser, setEncodedToken } = useAuth();
+    const [cartnumber, setCartNumber] = useState();
+    const [wishlistnumber, setWishlistNumber] = useState(0);
+    const { user, setUser, encodedToken, setEncodedToken } = useAuth();
     const navigate = useNavigate();
-    useEffect(() => {
-        if (user !== null) {
-            setLoginBtn("Logout");
-            // console.log("user at navbar : ", user)
-        }
-        else {
-            setLoginBtn("Login");
-        }
+
+    useEffect(async () => {
+        const res = await axios.get("/api/user/wishlist", {
+            headers: {
+                authorization: encodedToken
+            }
+        });
```

Comment:
this use effect will keep on running. you have to pass the secondary parameter. This will hang your browser see in console.

Reply:
Done, thanks for mentioning this.